src/screens/SignupScreen.js

```diff
@@ -0,0 +1,67 @@
+import React, {Component} from 'react';
+import {Redirect, withRouter} from "react-router";
+import {FormattedMessage, injectIntl} from "react-intl";
+import {endpoint} from "../api";
+import {xhrErrorToast, xhrSuccessToast} from "../components/XhrToast";
+import {connect} from "react-redux";
+import {login} from "../actions/sessionActions";
+
+class SignupScreen extends Component {
+  elements = {};
+
+  submit(event) {
+    const {match: {params}, intl, metadata, dispatch} = this.props;
+    event.preventDefault();
+
+    const data = {code: params.code};
+    Object.keys(this.elements).forEach(key => data[key] = this.elements[key].value);
```

Comment:
Lodash has a nice function called `mapValues` which might be prettier?

I think it would be the following (not tested):
```
const data = mapValues(this.elements, 'value');
```